File: spring_security/eg05_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/java/com/sc/sb/auth/repository/ScUserRoleRepository.java
package com.sc.sb.auth.repository;

import com.sc.sb.auth.entity.ScUserRole;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.CrudRepository;
import org.springframework.data.repository.query.Param;

import java.util.List;

public interface ScUserRoleRepository extends CrudRepository<ScUserRole, Long> {

    @Query("""
        select scRole.roleName from ScRole scRole
        join ScUserRole scUserRole on scRole.id = scUserRole.scRole.id
        where scUserRole.scUser.id = :scUserId
        """)
    List<String> findScRolesByScUserId(@Param("scUserId") Long scUserId);
}

File: spring_security/eg06_kotlin_sb3_oauth2_resource_server_jwt_symmetric_key/src/main/kotlin/com/sc/sb/auth/repository/ScUserRoleRepository.kt
package com.sc.sb.auth.repository

import com.sc.sb.auth.entity.ScUserRole
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param

interface ScUserRoleRepository : CrudRepository<ScUserRole, Long> {

    @Query(
        """
        select scRole.roleName from ScRole scRole
        join ScUserRole scUserRole on scRole.id = scUserRole.scRole.id
        where scUserRole.scUser.id = :scUserId"""
    )
    fun findScRolesByScUserId(@Param("scUserId") scUserId: Long): List<String>
}